core/beacon/types.go

```diff
@@ -30,16 +30,40 @@ import (
 
 // PayloadAttributesV1 structure described at https://github.com/ethereum/execution-apis/pull/74
 type PayloadAttributesV1 struct {
-	Timestamp             uint64         `json:"timestamp"     gencodec:"required"`
-	Random                common.Hash    `json:"prevRandao"        gencodec:"required"`
-	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient"  gencodec:"required"`
+	Timestamp             uint64         `json:"timestamp"             gencodec:"required"`
+	Random                common.Hash    `json:"prevRandao"            gencodec:"required"`
+	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient" gencodec:"required"`
+}
+
+func (a *PayloadAttributesV1) ToV2() *PayloadAttributesV2 {
```

Comment:
Please share any suggestions for improving this code. I thought about two other approaches:
1. Interface -- I could define an interface for these types which is the sum of all the versions. The downside is I believe I will need to then manually define the JSON encoding for each and implement a bunch of simple getters.
2. Single type -- I could extend the original type (and remove the V1) and then guard invalid combinations in the call sites. The downside for this is then that we always need to be careful about guarding against invalid configurations. Which may not be a big issue.

Reply:
I ended up trying 2) in f5c9b70.